### PR TITLE
Kotlin Cleanup in DeckPickerFloatingActionMenu.Kt

### PR DESCRIPTION
**Pull request by divyansh-dxn**
## Purpose / Description
Kotlin Cleanup

## Fixes
Fixes a part of [#10489](https://github.com/ankidroid/Anki-Android/issues/10489)

## Approach
1. Removed redundant type casts
2. Assigned types for some variables
3. Replaced null comparison with let

## How Has This Been Tested?
1. Passes all unit tests
2. Works as expected on Realme 6i and Pixel 4 Emulator

https://user-images.githubusercontent.com/69595691/158171656-e576be36-2b81-4edb-a27f-23bf2246014f.mp4

## Checklist
- [✓] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [✓] You have a descriptive commit message with a short title (first line, max 50 chars).
- [✓] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [✓] You have commented your code, particularly in hard-to-understand areas
- [✓] You have performed a self-review of your own code
- [✕] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [✕] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 1]
```diff
@@ -21,30 +21,26 @@ import android.widget.LinearLayout
...
+    private val mAddSharedLayout: LinearLayout = view.findViewById(R.id.add_shared_layout)
+    private val mAddDeckLayout: LinearLayout = view.findViewById(R.id.add_deck_layout)
+    private val mAddNoteLayout: LinearLayout = view.findViewById(R.id.add_note_layout)
     private val mFabBGLayout: View
-    private val mStudyoptionsFrame: View?
+    private val mStudyOptionsFrame: View?
     var isFABOpen = false
-    private var mFragmented = false
     private val mLinearLayout: LinearLayout
+
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:35]: could please revert this extra comment?

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 2]
```diff
@@ -21,30 +21,26 @@ import android.widget.LinearLayout
...
-    private val mAddSharedLayout: LinearLayout = view.findViewById<View>(R.id.add_shared_layout) as LinearLayout
-    private val mAddDeckLayout: LinearLayout = view.findViewById<View>(R.id.add_deck_layout) as LinearLayout
-    private val mAddNoteLayout: LinearLayout = view.findViewById<View>(R.id.add_note_layout) as LinearLayout
+    private val mFabMain: FloatingActionButton = view.findViewById(R.id.fab_main)
+    private val mAddSharedLayout: LinearLayout = view.findViewById(R.id.add_shared_layout)
+    private val mAddDeckLayout: LinearLayout = view.findViewById(R.id.add_deck_layout)
+    private val mAddNoteLayout: LinearLayout = view.findViewById(R.id.add_note_layout)
     private val mFabBGLayout: View
-    private val mStudyoptionsFrame: View?
+    private val mStudyOptionsFrame: View?
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:32]: nice

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 3]
```diff
@@ -111,15 +105,12 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
...
-        val addDeckButton = view.findViewById<View>(R.id.add_deck_action) as FloatingActionButton
+        val addNoteButton: FloatingActionButton = view.findViewById(R.id.add_note_action)
+        val addSharedButton: FloatingActionButton = view.findViewById(R.id.add_shared_action)
+        val addDeckButton: FloatingActionButton = view.findViewById(R.id.add_deck_action)
         mFabBGLayout = view.findViewById(R.id.fabBGLayout)
         mLinearLayout = view.findViewById(R.id.deckpicker_view) // Layout deck_picker.xml is attached here
-
-        // Check to see if the view was correctly located
-        mStudyoptionsFrame = view.findViewById(R.id.studyoptions_fragment)
-        mFragmented = mStudyoptionsFrame != null
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:122]: i don't think so you need to remove the `mFragmented` variable here, it plays role in this file, kindly revert this change.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 4]
```diff
@@ -21,30 +21,26 @@ import android.widget.LinearLayout
...
     private fun animationEnabled(): Boolean {
         val preferences = AnkiDroidApp.getSharedPrefs(deckPicker)
         return !preferences.getBoolean("safeDisplay", false)
     }
 
     private fun showFloatingActionMenu() {
         mLinearLayout.alpha = 0.5f
-        if (mFragmented) {
-            mStudyoptionsFrame!!.alpha = 0.5f
-        }
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:47]: > @KotlinCleanup("can be changed to view.findViewById<FloatingActionButton>")

as mentioned in kotlincleanup, you only need to change the view,

you are not supposed to change this bussiness logic here, revert this change.

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
@divyansh-dxn Thank you for the contributions!

there are a few extra changes that need to be reverted.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt (hunk 4 above)
Comment (divyansh-dxn) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:47]: Okay I'm on it

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt (hunk 3 above)
Comment (divyansh-dxn) [reply AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:122]: Since the comment describes `mFragmented`, it should be above `mFragmented` variable. Right?

**Review by Akshay0701 — APPROVED** [review]
Looks good, 

just remove that one extra line.

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 5]
```diff
@@ -43,7 +41,7 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
     private fun showFloatingActionMenu() {
         mLinearLayout.alpha = 0.5f
         if (mFragmented) {
-            mStudyoptionsFrame!!.alpha = 0.5f
+            mStudyOptionsFrame!!.alpha = 0.5f
         }
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:45]: a suggestion for null safety:
```suggestion
            mStudyOptionsFrame?.let { it.alpha = 0.5f } 
        }
```

**Review by Akshay0701 — CHANGES_REQUESTED** [review]
Thanks for the changes!

just one suggestion(open to you), rest looks good.

**Comment by divyansh-dxn** [comment]
@Akshay0701 , can you once again go through the file. In my opinion, the variable `mFragmented` looks redundant to me ( just asking ).

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 6]
```diff
@@ -77,7 +75,7 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
     fun closeFloatingActionMenu() {
         mLinearLayout.alpha = 1f
         if (mFragmented) {
-            mStudyoptionsFrame!!.alpha = 1f
+            mStudyOptionsFrame?.let { it.alpha = 0.5f }
```
Comment (Akshay0701) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:78]: its previous value was not 0.5f, kindly change it to `1f`

**Review by Akshay0701 — CHANGES_REQUESTED** [review]

**Comment by Akshay0701** [comment]
> @Akshay0701, can you once again go through the file. In my opinion, the variable `mFragmented` looks redundant to me ( just asking ).

You're right, as we are handling null checks in our code, I don't see any point to keep `mFragmented` variable in this file.

**Comment by divyansh-dxn** [comment]
> > @Akshay0701, can you once again go through the file. In my opinion, the variable `mFragmented` looks redundant to me ( just asking ).
> 
> You're right, as we are handling null checks in our code, I don't see any point to keep `mFragmented` variable in this file.

Looks funny, that is what I had done in the first commit 🥲

**Review by Akshay0701 — APPROVED** [review]

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 7]
```diff
@@ -111,18 +109,19 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
...
+        val addNoteButton: FloatingActionButton = view.findViewById(R.id.add_note_action)
+        val addSharedButton: FloatingActionButton = view.findViewById(R.id.add_shared_action)
+        val addDeckButton: FloatingActionButton = view.findViewById(R.id.add_deck_action)
         mFabBGLayout = view.findViewById(R.id.fabBGLayout)
         mLinearLayout = view.findViewById(R.id.deckpicker_view) // Layout deck_picker.xml is attached here
-
-        // Check to see if the view was correctly located
-        mStudyoptionsFrame = view.findViewById(R.id.studyoptions_fragment)
-        mFragmented = mStudyoptionsFrame != null
+        mStudyOptionsFrame = view.findViewById(R.id.studyoptions_fragment)
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:117]: These 3 should be able to be moved to the `val` properties

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 8]
```diff
@@ -111,18 +109,19 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
...
         val addNoteLabel = view.findViewById<TextView>(R.id.add_note_label)
         val addSharedLabel = view.findViewById<TextView>(R.id.add_shared_label)
         val addDeckLabel = view.findViewById<TextView>(R.id.add_deck_label)
+
+        // Check to see if the view was correctly located
+        mFragmented = mStudyOptionsFrame != null
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:123]: Remove this variable entirely, OR: make it a `get() = mStudyOptionsFrame != null`, but there's no need to use it in this class currently

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 9]
```diff
@@ -43,7 +41,7 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
     private fun showFloatingActionMenu() {
         mLinearLayout.alpha = 0.5f
         if (mFragmented) {
-            mStudyoptionsFrame!!.alpha = 0.5f
+            mStudyOptionsFrame?.let { it.alpha = 0.5f }
         }
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:45]: remove the `if (mFragmented) {`

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 10]
```diff
@@ -21,17 +21,15 @@ import android.widget.LinearLayout
...
-    private val mAddNoteLayout: LinearLayout = view.findViewById<View>(R.id.add_note_layout) as LinearLayout
+    private val mFabMain: FloatingActionButton = view.findViewById(R.id.fab_main)
+    private val mAddSharedLayout: LinearLayout = view.findViewById(R.id.add_shared_layout)
+    private val mAddDeckLayout: LinearLayout = view.findViewById(R.id.add_deck_layout)
+    private val mAddNoteLayout: LinearLayout = view.findViewById(R.id.add_note_layout)
     private val mFabBGLayout: View
-    private val mStudyoptionsFrame: View?
+    private val mStudyOptionsFrame: View?
     var isFABOpen = false
     private var mFragmented = false
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:34]: Change to (psedocode)

```
Suppress("unused") 
val mFragmented : Boolean = get() mStudyOptionsFrame != null
```

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 11]
```diff
@@ -77,7 +75,7 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
     fun closeFloatingActionMenu() {
         mLinearLayout.alpha = 1f
         if (mFragmented) {
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:77]: Remove the `if (mFragmented)`

**Review by david-allison — COMMENTED** [review]
Looks good! I think we can do better

File: AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt [hunk 12]
```diff
@@ -28,21 +28,24 @@ class DeckPickerFloatingActionMenu(view: View, private val deckPicker: DeckPicke
...
-    private val mFabBGLayout: View
-    private val mStudyOptionsFrame: View?
+    private val mFabBGLayout: View = view.findViewById(R.id.fabBGLayout)
+    private val mLinearLayout: LinearLayout = view.findViewById(R.id.deckpicker_view) // Layout deck_picker.xml is attached here
+    private val mStudyOptionsFrame: View? = view.findViewById(R.id.studyoptions_fragment)
     var isFABOpen = false
-    private var mFragmented = false
-    private val mLinearLayout: LinearLayout
+
+    // Check to see if the view was correctly located
```
Comment (david-allison) [inline AnkiDroid/src/main/java/com/ichi2/anki/DeckPickerFloatingActionMenu.kt:36]:
```suggestion
```

**Review by david-allison — APPROVED** [review]
LGTM. Thanks!
